examples/gno.land/p/demo/diff/diff.gno

```diff
@@ -0,0 +1,151 @@
+package diff
+
+import (
+	"strings"
+)
+
+type EditType int
+
+const (
+	EditKeep EditType = iota
+	EditInsert
+	EditDelete
+)
+
+type Edit struct {
+	Type EditType
+	Char rune
+}
+
+func MyersDiff(old, new string) []Edit {
+	oldRunes, newRunes := []rune(old), []rune(new)
+	n, m := len(oldRunes), len(newRunes)
+
+	if n == 0 && m == 0 {
+		return []Edit{}
+	}
+
+	// old is empty
+	if n == 0 {
+		edits := make([]Edit, m)
+		for i, r := range newRunes {
+			edits[i] = Edit{Type: EditInsert, Char: r}
+		}
+		return edits
+	}
+
+	if m == 0 {
+		edits := make([]Edit, n)
+		for i, r := range oldRunes {
+			edits[i] = Edit{Type: EditDelete, Char: r}
+		}
+		return edits
+	}
+
+	max := n + m
+	v := make([]int, 2*max+1)
+	var trace [][]int
+
+	for d := 0; d <= max; d++ {
+		for k := -d; k <= d; k += 2 {
+			var x int
+			if k == -d || (k != d && v[max+k-1] < v[max+k+1]) {
+				x = v[max+k+1]
+			} else {
+				x = v[max+k-1] + 1
+			}
+			y := x - k
+
+			for x < n && y < m && oldRunes[x] == newRunes[y] {
+				x++
+				y++
+			}
+
+			v[max+k] = x
+
+			if x == n && y == m {
+				trace = append(trace, append([]int(nil), v...))
+				goto endSearch
```

Comment:
```suggestion
				break search
```

Nit pick: avoid the `goto` and use `break LABEL` instead. It's not just for esthetic reason, it will also help static analysis programs to better analyse the flow, when we'll use them to assess gno code.